ocaml/lambda/lambda.ml

```diff
@@ -1713,6 +1732,10 @@ let primitive_may_allocate : primitive -> alloc_mode option = function
   | Pmakefloatblock (_, m) -> Some m
   | Pmakeufloatblock (_, m) -> Some m
   | Pmakemixedblock (_, _, _, m) -> Some m
+  | Preuseblock { mode } -> Some mode
```

Comment:
But these new primitives can't allocate, can they? It's one of the reasons I don't think the `mode` is necessary.

Reply:
The `mode` will soon also contain uniqueness information that informs us whether the new block will itself be used uniquely. One worry then might be if the optimizer can lift out `Preuseblock`s that are unique? (In fact, it can't: it could only do that if the cell we reuse is statically known, but that cell will have been marked as unique and can not be lifted out). Furthermore, we sometimes handle `Pmakeblock` and `Preuseblock` similarly (eg. in `tmc.ml`) and it might be helpful if they have the same fields.

Reply:
This line is in the function `primitive_may_allocate`, and returning `Some mode` makes it sound like it might allocate. (Perhaps my comment about the `mode` buried my main point.)

One worry: it _does_ allocate in bytecode. Does that matter here?

Reply:
It would be sound to keep the `Some mode` here, since `primitive_may_allocate` is only used in `transl_primitive` where this value causes strictly more checks. Since we have decided to map `Preuseblock` to `Pmakeblock` in the backend for the moment, I think we should stick with this code for now. However, I will add a CR to make all of these return `None` once the backend supports actual overwriting. The bytecode interpreter does not use stack allocation so this check is irrelevant there.